### PR TITLE
fix(editor3): fix paste performance when suggesting

### PR DESCRIPTION
**Pull request by petrjasek**
handle whole pasted content at once, not each character separate.

SDFID-361

### CODE REVIEW

File: scripts/core/editor3/reducers/suggestions.jsx [hunk 1]
```diff
@@ -206,26 +247,41 @@ const createChangeBlockStyleSuggestion = (state, {blockType, data}) => {
...
+    // push new content
+    editorState = EditorState.push(editorState, mergedContent, 'insert-fragment');
 
-    return saveEditorStatus(state, editorState, 'change-block-data');
+    // store current selection for later
+    const finalSelection = editorState.getSelection();
+
+    // select pasted content
+    editorState = EditorState.acceptSelection(editorState, editorState.getSelection().merge({
+        anchorKey: initialSelection.getAnchorKey(),
```
Comment (ioanpocol) [inline scripts/core/editor3/reducers/suggestions.jsx]: Is this working well when the initial selection is backward?
Usually I use getStartKey/Offset and getEndKey/Offset and isBackward: false.

File: scripts/core/editor3/reducers/suggestions.jsx [hunk 2]
```diff
@@ -195,6 +195,47 @@ const createChangeBlockStyleSuggestion = (state, {blockType, data}) => {
...
+            }
+        );
+
+        // remove any entities
+        block.findEntityRanges(
+            () => true,
+            (start, end) => {
+                const selection = getSelection(block, start, end);
+
+                output = Modifier.applyEntity(output, selection, null);
```
Comment (ioanpocol) [inline scripts/core/editor3/reducers/suggestions.jsx]: Not sure if we have to delete the Entity, for example when you copy  a link from the same editor.

**Comment by ioanpocol** [comment]
:+1:

File: scripts/core/editor3/reducers/suggestions.jsx (hunk 1 above)
Comment (petrjasek) [reply scripts/core/editor3/reducers/suggestions.jsx]: I think it doesn't, will push an update

File: scripts/core/editor3/reducers/suggestions.jsx (hunk 2 above)
Comment (petrjasek) [reply scripts/core/editor3/reducers/suggestions.jsx]: yep we could keep that, I was testing with external links and those didn't work so I've rather disabled it for now

**Pull request merged**